### PR TITLE
Prevent the dialog from showing when the state of the activity/fragment has already passed onSavedInstanceState.

### LINKED ISSUE #435: AuthenticationFragment.showError: IllegalStateException - Can not perform this action after onSaveInstanceState
New in 1.7.2 according to Crashlytics, but seems similar to #367. 90 crashes for 62 users so far.

https://console.firebase.google.com/u/0/project/home-assistant-mobile-apps/crashlytics/app/android:io.homeassistant.companion.android/issues/790b513ff4614a3a72e6f7e65f22472b?time=last-seven-days&sessionId=5E4892B200CB0001509468EC4ED1DE34_DNE_0_v2

```
Fatal Exception: java.lang.IllegalStateException: Can not perform this action after onSaveInstanceState
       at androidx.fragment.app.FragmentManagerImpl.checkStateLoss(FragmentManagerImpl.java:1536)
       at androidx.fragment.app.FragmentManagerImpl.enqueueAction(FragmentManagerImpl.java:1558)
       at androidx.fragment.app.FragmentManagerImpl.popBackStack(FragmentManagerImpl.java:247)
       at io.homeassistant.companion.android.onboarding.authentication.AuthenticationFragment.showError(AuthenticationFragment.kt:115)
       at io.homeassistant.companion.android.onboarding.authentication.AuthenticationFragment$onCreateView$$inlined$apply$lambda$1.onReceivedError(AuthenticationFragment.kt:70)
       at Ke0.handleMessage(PG:133)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6944)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
```

### PR DESCRIPTION
**Pull request by Timmmeeeh**
Prevent the dialog from showing when the state of the activity/fragment has already passed onSavedInstanceState. This fixes #435.

### CODE REVIEW

**Comment by homeassistant** [comment]
Hello @Timmmeeeh,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`7599ec9423ad0cb7caedbac46d3545e52ccecf39`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant-android&#x2F;pull&#x2F;450&#x2F;commits&#x2F;7599ec9423ad0cb7caedbac46d3545e52ccecf39) - This commit has something that looks like an email address (timmeeeeh@outlook.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Comment by homeassistant** [comment]
Hi @Timmmeeeh, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-android#450).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt [hunk 1]
```diff
@@ -329,7 +330,7 @@ class WebViewActivity : AppCompatActivity(), io.homeassistant.companion.android.
     }
 
     override fun showError(isAuthenticationError: Boolean) {
-        if (isShowingError || isFinishing)
+        if (isShowingError || !isStarted)
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt]: I don't think we want to remove the check for `isFinishing`

File: app/src/main/java/io/homeassistant/companion/android/onboarding/authentication/AuthenticationFragment.kt [hunk 2]
```diff
@@ -103,8 +104,8 @@ class AuthenticationFragment : Fragment(), AuthenticationView {
     }
 
     override fun showError(message: Int) {
-        if (activity?.isFinishing != false) {
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/onboarding/authentication/AuthenticationFragment.kt]: I don't think we want to remove the check for `isFinishing`

Comment (Timmmeeeh) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/authentication/AuthenticationFragment.kt]: isFinishing is only useful when, the user hits the back button or Activity#finish() has explicitly been called. There are more ways an activity or fragment can get destroyed. So I think using isFinishing in this situation doesn't suffice. [isStarted](https://developer.android.com/reference/kotlin/androidx/lifecycle/Lifecycle.State#started) will already be false when it reached onPause.

Comment (JBassett) [reply app/src/main/java/io/homeassistant/companion/android/onboarding/authentication/AuthenticationFragment.kt]: Totally missed that it covered that case as well.

**Pull request merged**